docs/runbooks/csp-health-monitor.md

```diff
@@ -0,0 +1,224 @@
+```
+
+## AWS Issues
+
+### Symptom: AccessDeniedException Errors
+
+**Logs show:**
+```
+Error while fetching maintenance events: operation error Health: DescribeEvents, https response error StatusCode: 403, AccessDeniedException
+```
+
+### Verification Steps
+
+1. **Check IAM policy is attached to role:**
+
+```bash
+aws iam list-attached-role-policies \
+    --role-name <CLUSTER_NAME>-nvsentinel-health-monitor-assume-role-policy
+```
+
+Expected output should show `CSPHealthMonitorPolicy` attached.
+
+2. **Check IAM role trust policy:**
+
+```bash
+aws iam get-role \
+    --role-name <CLUSTER_NAME>-nvsentinel-health-monitor-assume-role-policy \
+    --query 'Role.AssumeRolePolicyDocument'
+```
+
+Expected: Trust policy should reference the correct EKS OIDC provider and `system:serviceaccount:nvsentinel:csp-health-monitor`.
+
+3. **Check ServiceAccount annotation:**
+
+```bash
+kubectl get serviceaccount csp-health-monitor -n nvsentinel -o jsonpath='{.metadata.annotations.eks\.amazonaws\.com/role-arn}'
+```
+
+Expected output: `arn:aws:iam::<ACCOUNT_ID>:role/<CLUSTER_NAME>-nvsentinel-health-monitor-assume-role-policy`
+
+### Resolution
+
+If IAM policy is not attached:
+
+```bash
+ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+
+aws iam attach-role-policy \
+    --role-name <CLUSTER_NAME>-nvsentinel-health-monitor-assume-role-policy \
+    --policy-arn arn:aws:iam::${ACCOUNT_ID}:policy/CSPHealthMonitorPolicy
+```
+
+If the role ARN doesn't match Helm values, update `configToml.clusterName` and redeploy.
+
+### Test Permissions Manually
+
+```bash
+aws health describe-events --filter "services=EC2" --max-items 1
+```
+
+## Node Mapping Failures
+
+### Symptom: Events Detected but Nodes Not Quarantined
+
+**Logs show:**
+```
+No Kubernetes node found matching GCP numeric instance ID
+Instance ID not found in node map
+```
+
+### Verification Steps
+
+1. **Check nodes have providerID set:**
+
+```bash
+kubectl get nodes -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.spec.providerID}{"\n"}{end}'
+```
+
+Expected:
+- GCP: `gce://<project-id>/<zone>/<instance-name>`
+- AWS: `aws:///<availability-zone>/<instance-id>`
+
+2. **Check GCP node annotations (GCP only):**
+
+```bash
+kubectl get nodes -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.annotations.container\.googleapis\.com/instance_id}{"\n"}{end}'
+```
+
+3. **Check RBAC permissions:**
+
+```bash
+kubectl auth can-i list nodes --as=system:serviceaccount:nvsentinel:csp-health-monitor
+```
+
+Expected: `yes`
+
+### Resolution
+
+If nodes missing `providerID`, the kubelet configuration may be incorrect. Check node registration and cloud provider integration.
+
+If RBAC is missing, verify the ClusterRole and ClusterRoleBinding were created by the Helm chart:
+
+```bash
+kubectl get clusterrole csp-health-monitor
+kubectl get clusterrolebinding csp-health-monitor
+```
+
+## General Diagnostics
```

Comment:
Let's remove this section. Let's only keep the IAM specific items in this runbook. We can also rename the runbook to be specific to IAM for CSP health monitor